package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -679,90 +679,136 @@ def _get_dh_pairs(self):
 
         return donors, hydrogens
 
-    def _filter_atoms(self, donors, acceptors):
-        """Create a mask to filter donor, hydrogen and acceptor atoms.
-
-        This can be used to consider only hydrogen bonds between two or more
-        specified groups.
-
-        Groups are specified with the `between` keyword when creating the
-        HydrogenBondAnalysis object.
-
-           Returns
-           -------
-           mask: np.ndarray
-
-
-        .. versionchanged:: 2.5.0
-           Change return value to a mask instead of separate AtomGroups.
-``
-        """
-
-        mask = np.full(donors.n_atoms, fill_value=False)
-        for group1, group2 in self.between_ags:
-
-            # Find donors in G1 and acceptors in G2
-            mask[
-                    np.logical_and(
-                        np.isin(donors.indices, group1.indices),
-                        np.isin(acceptors.indices, group2.indices)
-                    )
-            ] = True
-
-            # Find acceptors in G1 and donors in G2
-            mask[
-                np.logical_and(
-                    np.isin(acceptors.indices, group1.indices),
-                    np.isin(donors.indices, group2.indices)
-                )
-            ] = True
-
-        return mask
-
-
     def _prepare(self):
         self.results.hbonds = [[], [], [], [], [], []]
-
-
+ 
     def _single_frame(self):
-
         box = self._ts.dimensions
 
         # Update donor-hydrogen pairs if necessary
         if self.update_selections:
             self._donors, self._hydrogens = self._get_dh_pairs()
 
-        # find D and A within cutoff distance of one another
-        # min_cutoff = 1.0 as an atom cannot form a hydrogen bond with itself
-        d_a_indices, d_a_distances = capped_distance(
-            self._donors.positions,
-            self._acceptors.positions,
-            max_cutoff=self.d_a_cutoff,
-            min_cutoff=1.0,
-            box=box,
-            return_distances=True,
-        )
+        if self.between_ags is not None:
```

Comment:
please could you move everything in this `if` block into a separate function to make `_single_frame` a little easier to read?